ipv4/setnode.go

```diff
@@ -296,3 +297,30 @@ func (me *setNode) height() int {
 func (me *setNode) Walk(callback func(Prefix, interface{}) bool) bool {
 	return (*trieNode)(me).Walk(callback)
 }
+
+func (me *setNode) FindSmallestContainingPrefix(length uint32) (Prefix, error) {
+	if me == nil || length < me.Prefix.length {
+		return Prefix{}, fmt.Errorf("cannot find containing prefix")
+	}
+	if length == me.Prefix.length {
+		return me.Prefix, nil
```

Comment:
There is an `is_active` boolean in the node that maybe I forgot to consider. I'll try to craft a test case that would hit this when the node has is_active=false to see if gets an incorrect result.